packages/carbon-web-components/src/components/slug/slug.scss

```
@@ -315,6 +315,121 @@ $colorMap: (
   }
 }
 
+// caret styles for auto align
+/* stylelint-disable-next-line no-duplicate-selectors */
+:host(
+    #{$prefix}-slug[autoalign]:not([kind='hollow']):not([dot-type='hollow'])
+  ) {
+  .#{$prefix}--popover-caret::before,
+  .#{$prefix}--popover-caret::after {
+    display: none;
+  }
+
+  .#{$prefix}--popover-caret {
+    block-size: 10px;
+    clip-path: polygon(0 0, 0% 100%, 100% 0);
+    inline-size: 10px;
+  }
+}
+
+/* stylelint-disable-next-line no-duplicate-selectors */
+:host(
+    #{$prefix}-slug[autoalign]:not([kind='hollow']):not([dot-type='hollow'])
+  ) {
+  &[alignment='top'],
+  &[alignment='top-left'],
+  &[alignment='top-right'],
+  &[alignment='top-start'],
+  &[alignment='top-end'] {
```

could these styles be? 

```suggestion
  &[alignment^='top'], {
```

---

Ah yes!